main/src/ca/uwaterloo/flix/api/lsp/provider/Visitor.scala

```diff
@@ -0,0 +1,850 @@
+    (!(posLine == loc.endLine && posCol >= loc.endCol)) // geq because end column is exclusive
+  }
+
+  def inside(loc1: SourceLocation, loc2: SourceLocation): Boolean = {
+    loc1.source == loc2.source &&
+    (loc1.beginLine >= loc2.beginLine) &&
+    (loc1.endLine <= loc2.endLine) &&
+    !(loc2.beginLine == loc1.beginLine && loc2.beginCol > loc1.beginCol) &&
+    !(loc1.endLine == loc2.endLine && loc1.endCol > loc2.endCol)
+  }
+
+  private def visitEnum(enm: Enum)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(enm.loc)) { return }
+
+    c.consumeEnum(enm)
+
+    visitAnnotations(enm.ann)
+    // TODO visit modifiers?
+    enm.tparams.foreach(visitTypeParam)
+    visitDeriveList(enm.derives)
+    enm.cases.map(_._2).foreach(visitCase)
+  }
+
+  private def visitDeriveList(deriveList: Derivations)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(deriveList.loc)) { return }
+    c.consumeDeriveList(deriveList)
+    deriveList.traits.foreach(visitDerive)
+  }
+
+  private def visitDerive(derive: Derivation)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(derive.loc)) { return }
+    c.consumeDerive(derive)
+  }
+
+  private def visitCase(cse: Case)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(cse.loc)) { return }
+    c.consumeCase(cse)
+    // TODO visit scheme?
+  }
+
+  private def visitInstance(ins: Instance)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(ins.loc)) { return }
+
+    visitAnnotations(ins.ann)
+    visitTraitSymUse(ins.trt)
+    ins.tconstrs.foreach(visitTraitConstraint)
+    ins.assocs.foreach(visitAssocTypeDef)
+    ins.defs.foreach(visitDef)
+  }
+
+  private def visitTraitSymUse(symUse: TraitSymUse)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(symUse.loc)) { return }
+    c.consumeTraitSymUse(symUse)
+  }
+
+  private def visitTraitConstraint(tc: TraitConstraint)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(tc.loc)) { return }
+
+    c.consumeTraitConstraint(tc)
+    visitTraitConstraintHead(tc.head)
+    visitType(tc.arg)
+  }
+
+  private def visitTraitConstraintHead(tcHead: TraitConstraint.Head)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(tcHead.loc)) { return }
+    c.consumeTraitConstraintHead(tcHead)
+  }
+
+  private def visitAssocTypeDef(tdefn: AssocTypeDef)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(tdefn.loc)) { return }
+    c.consumeAssocTypeDef(tdefn)
+    visitAssocTypeSymUse(tdefn.sym)
+    visitType(tdefn.arg)
+  }
+
+  private def visitAssocTypeSymUse(symUse: AssocTypeSymUse)(implicit a: Acceptor, c: Consumer): Unit = {
+    if (!a.accept(symUse.loc)) { return }
+    c.consumeAssocTypeSymUse(symUse)
+  }
+
+  private def visitSig(sig: Sig)(implicit a: Acceptor, c: Consumer): Unit = {
+    // TODO fix hack. This hack is necessary because a signature currently does not have a location.
```

Comment:
Refactor into an actual comment.

Reply:
and consider a PR to fix the issue.

Reply:
what do you by refactoring into actual comment? Isn't it an actual comment? Just that the comment should be more clear?